spynnaker/pyNN/models/spike_source/spike_source_poisson_machine_vertex.py

```diff
@@ -421,14 +424,16 @@ def _write_poisson_rates(self, spec):
                  _u3232_to_uint64(item['starts']),
                  _u3232_to_uint64(item['durations']))
                 )[0]
-            data_items.extend([[count], [len(items)], [0],
-                               numpy.ravel(items).view("uint32")])
+            data_to_add = [[count], [len(items)], [0],
+                           numpy.ravel(items).view("uint32")]
+            data_items.extend(data_to_add)
             n_items += 1
+            data_size += count * (3 + len(data_to_add[-1]))
```

Comment:
Note this could simply use get_expander_rates_bytes, which would then match up with the pre-computed value

Reply:
Done and fixed the calculation of get_expander_rates_bytes at the same time.